### PR TITLE
Enabling different types of payments methods

### PR DESCRIPTION
**Pull request by svenpopping**
### Description
Currently payments can only handle iDeal payments. However this year there are a lot more international students that are not able to use iDeal. This PR adds the payments methods Credit Card, Sofort and Paypal. It also makes it possible to easily extend this list.

### Todos
- [x] Tests
- [x] Documentation
- [x] Add migration
- [x] Confirmation by the board about payments methods
- [x] Enabling payments methods in Mollie profile

### Steps to Test or Reproduce
Run the tests

### Deployment requirements
- This PR includes a Migration for the class Order, so after deploy check if migration is successful.
- Added payment methods should be enabled in the Mollie profile

### CODE REVIEW

File: src/main/java/ch/wisv/payments/model/MollieMethodEnum.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import net.objecthunter.exp4j.Expression;
+import net.objecthunter.exp4j.ExpressionBuilder;
+
+/**
+ * MollieMethodEnum.
+ */
+public enum MollieMethodEnum {
+
+    IDEAL("ideal", new ExpressionBuilder("x + 0.35")),
+    CREDIT_CARD("creditcard", new ExpressionBuilder("1.03388 * x + 0.35")),
```
Comment (FrancisBehnen) [inline src/main/java/ch/wisv/payments/model/MollieMethodEnum.java]: 3.388% transaction fee, it was only 2.8% for non-eu cards right?

File: src/main/java/ch/wisv/payments/model/OrderRequest.java [hunk 2]
```diff
@@ -14,11 +13,15 @@
...
     @NotEmpty
     String name;
+
     @NotEmpty
     String email;
 
     @URL
     String returnUrl;
 
+    // Default is set to IDEAL, as old implementations do not include the method type in there request.
```
Comment (FrancisBehnen) [inline src/main/java/ch/wisv/payments/model/OrderRequest.java]: Their* :p

File: src/main/resources/db/migration/V20171213__payment_methods.sql [hunk 3]
```diff
@@ -0,0 +1 @@
+ALTER TABLE orders ADD method integer NOT NULL DEFAULT 0;
```
Comment (FrancisBehnen) [inline src/main/resources/db/migration/V20171213__payment_methods.sql]: Missing eof new line

File: src/test/java/ch/wisv/payments/model/MollieMethodEnumTest.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        assertEquals(0.35, method.calculateCostIncludingTransaction(0));
+    }
+
+    public void testPaypalCost() {
+        MollieMethodEnum method = MollieMethodEnum.PAYPAL;
+
+        assertEquals(1.12, method.calculateCostIncludingTransaction(1));
+        assertEquals(0.12, method.calculateCostIncludingTransaction(0));
+    }
+}
```
Comment (FrancisBehnen) [inline src/test/java/ch/wisv/payments/model/MollieMethodEnumTest.java]: Missing eof new line

**Review by FrancisBehnen — APPROVED** [review]

**Comment by skamoen** [comment]
If you leave the payment method option blank, mollie will send you to a method selection screen automatically, containing enabled methods from the mollie dashboard

File: src/main/java/ch/wisv/payments/model/MollieMethodEnum.java (hunk 1 above)
Comment (svenpopping) [reply src/main/java/ch/wisv/payments/model/MollieMethodEnum.java]: 3.3388% is including taxes, and I took the highest of the two.

Comment (FrancisBehnen) [reply src/main/java/ch/wisv/payments/model/MollieMethodEnum.java]: Ah right, taxes..

**Comment by FrancisBehnen** [comment]
@skamoen does it then automatically add the transaction fees?

**Comment by svenpopping** [comment]
Guess not!

**Comment by FrancisBehnen** [comment]
Just had a little discussion with Ricardo and @dvantetering, we realised that, unlike iDEAL, creditcard transactions area reversible. To serve eu-students we can add non-reversible payments like SOFORT (and bancontact maybe), but we want to block reversible payments for now.

File: src/main/java/ch/wisv/payments/model/MollieMethodEnum.java [hunk 5]
```diff
@@ -9,10 +9,20 @@
  */
 public enum MollieMethodEnum {
 
+    /**
+     * Bancontact (Belgium).
+     */
+    BANCONTACT("bancontact", new ExpressionBuilder("1.01815 * x + 0.3025")),
```
Comment (FrancisBehnen) [inline src/main/java/ch/wisv/payments/model/MollieMethodEnum.java]: Maybe add the breakdown of this calculation in the comments, so it's clearer that the mollie website is excl. VAT and this incl. VAT

**Review by FrancisBehnen — APPROVED** [review]

**Pull request merged**